### PR TITLE
Added DataCite support

### PR DESCRIPTION
**Pull request by vjrj**
I added DataCite support using the [gbif-doi](https://github.com/gbif/gbif-doi) library (thanks to @timrobertson100 for the suggestion and for help providing me some test datacite account).

I tested minting test DOIs using `api.test.datacite.org` but in short I'll try with the production API in:
https://doi.gbif.es

I'll also send an additional PR to `ala-install` to allow the configuration of the new variables.

### CODE REVIEW

File: grails-app/services/au/org/ala/doi/DoiService.groovy [hunk 1]
```diff
@@ -24,8 +25,9 @@ import static au.org.ala.doi.util.Utils.isUuid
 class DoiService extends BaseDataAccessService {
 
     GrailsApplication grailsApplication
-    AndsService andsService
-    MockService mockService
+    AndsService andsService = new AndsService()
```
Comment (sbearcsiro) [inline grails-app/services/au/org/ala/doi/DoiService.groovy:28]: Don't `new` these Service classes, leave them to Grails to inject

**Comment by vjrj** [comment]
Off course, thanks for the review.

Sorry, I'm a java guy using a java library here, so many times I missed groovy. I added and deleted a lot of semicolons :-P

File: grails-app/services/au/org/ala/doi/providers/DataCiteService.groovy [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+import org.gbif.doi.service.datacite.RestJsonApiDataCiteService
+
+import static org.gbif.doi.metadata.datacite.DataCiteMetadata.*
+import static org.gbif.doi.metadata.datacite.DataCiteMetadata.RightsList.Rights
+import static org.gbif.doi.metadata.datacite.DataCiteMetadata.Titles.Title
+
+class DataCiteService extends DoiProviderService {
+    DataCiteClient dataCiteClient
+    RestJsonApiDataCiteService restDataCiteService
+    ClientConfiguration clientConf = ClientConfiguration.builder().withBaseApiUrl(
```
[the remark this answers is not in the document]
Comment (vjrj) [reply grails-app/services/au/org/ala/doi/providers/DataCiteService.groovy]: Done

File: grails-app/services/au/org/ala/doi/providers/DataCiteService.groovy [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+            def rightsList = new RightsList()
+            metadata.rights.each {
+                def rights = new Rights()
+                rights.setValue((String) it)
+                rightsList.rights.add(rights)
+            }
+            dcMetadata.setRightsList(rightsList)
+        }
+
+        // FIXME put lang in config
```
[the remark this answers is not in the document]
Comment (vjrj) [reply grails-app/services/au/org/ala/doi/providers/DataCiteService.groovy]: Done

**Comment by vjrj** [comment]
Hi @sbearcsiro I followed all your comments. All tests, etc worked. But in some point for some reason, the pages are unstyled, without header/footer. This also happened to me testing some recents versions and snapshots from nexus via ala-install. Can you share your configuration skin and  headerAndFooter configuration?

2020-09-05-09:06:48-screenshot

**Comment by brucehyslop** [comment]
A new issue #57 has been created to address the broken style.

**Comment by brucehyslop** [comment]
broken style has been fixed. 

@sbearcsiro are you happy with the requested changes?

**Comment by nickdos** [comment]
Bump @sbearcsiro

**Pull request merged**